packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -1173,12 +1173,27 @@ class RenderEditable extends RenderBox {
 
   /// Move selection to the location of the last tap down.
   void selectPosition({@required SelectionChangedCause cause}) {
+    selectPositionAt(from: _lastTapDownPosition, cause: cause);
+  }
+
+  /// Select text between the global positions [from] and [to].
+  void selectPositionAt({@required Offset from, Offset to, @required SelectionChangedCause cause}) {
     assert(cause != null);
     _layoutText(constraints.maxWidth);
-    assert(_lastTapDownPosition != null);
     if (onSelectionChanged != null) {
-      final TextPosition position = _textPainter.getPositionForOffset(globalToLocal(_lastTapDownPosition));
-      onSelectionChanged(TextSelection.fromPosition(position), this, cause);
+      final TextPosition fromPosition = _textPainter.getPositionForOffset(globalToLocal(from + -_paintOffset));
```

Comment:
why not `from - _paintOffset`?

Reply:
I don't really know. Changed all the previous usages in this file too.